### ISSUE TITLE
Use flow-typed for builtin types instead of storing them in flow itself

### ISSUE THREAD

**Issue by jedwards1211** [facebook/flow#5248]
Some of us are unfortunately stuck on old versions of flow because of the dramatic effort it would take to bring dozens of interrelated private packages in our orgs up-to-date and fix the resulting new flow errors.  (I've been working to try to automate that process but there are still difficulties, like the defs packaged with Immutable.js not quite working with the latest version of flow.)

It becomes a significant problem when the builtin node module defs in an old version of flow are out of date.

If the type defs for builtin node modules were stored in flow-typed instead, it would enable us to upgrade the defs without having to upgrade flow (we might have to PR upgrades to the defs for an old flow version to flow-typed, but that would be better than nothing).

**Comment by aikar** [comment]
You can download the latest file and throw it in flow-typed/node.js

**Comment by jedwards1211** [comment]
@aikar that's true...but would it actually override that defined in flow itself?

**Comment by SamChou19815** [comment]
Most of the libdef has been moved to flow-typed.
